### PR TITLE
Better preprocessor and 3 bugfixes

### PR DESCRIPTION
**Pull request by olydis**
Bugfixes:
- had a throwing `catch` block in `SwaggerParser` that prevented us from seeing meaningful JSON parsing exceptions
- C# generator emitted `Stream.Null` without any usings
- #1716

### CODE REVIEW

File: src/generator/AutoRest.CSharp/CodeNamerCs.cs [hunk 1]
```diff
@@ -80,6 +80,10 @@ public override string EscapeDefaultValue(string defaultValue, IModelType type)
                 {
                     return "new " + type.Name + "()";
                 }
+                if (type is EnumType && (type as EnumType).ModelAsString)
```
Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp/CodeNamerCs.cs]: Did you do a test run with the compare script? I'm nervous about what happens on all the generators...

Comment (olydis) [inline src/generator/AutoRest.CSharp/CodeNamerCs.cs]: something like another code location relying on the thing being unquoted and adding its own set of quotes? sounds likely, will do :-)

Comment (olydis) [inline src/generator/AutoRest.CSharp/CodeNamerCs.cs]: empty diff for me!

Comment (fearthecowboy) [inline src/generator/AutoRest.CSharp/CodeNamerCs.cs]: Awesome!

**Pull request merged**